// const db = require("./db");
// async function registrarUsuario(nick) {
// return await db.insertOne("usuario",{"nick": nick});
// }


// let buscarUsuario = async (iduser)=>{
// 	let user = await db.findOne("usuario",iduser);
// 	return user;
// }

// let alterarUsuario = async (user)=>{
// 	return await db.updateOne("usuario", user,{_id:user._id});
// }


  
// module.exports = {registrarUsuario, buscarUsuario, alterarUsuario}


const db = require("./db");

async function registrarUsuario(nick) {
	console.log("Tentando registrar usu치rio com nick:", nick);
	try {
	  const result = await db.insertOne("usuario", {"nick": nick});
	  console.log("Resultado do registro:", result);
	  return result;
	} catch (error) {
	  console.error("Erro ao registrar usu치rio:", error);
	  throw error;
	}
  }


  async function buscarUsuario(iduser) {
	console.log("Tentando buscar usu치rio com id:", iduser);
	try {
	  const result = await db.findOne("usuario", iduser);
	  console.log("Resultado da busca:", result);
	  return result;
	} catch (error) {
	  console.error("Erro ao buscar usu치rio:", error);
	  throw error;
	}
  }

let alterarUsuario = async (user)=>{
	return await db.updateOne("usuario", user,{_id:user._id});
}


  
module.exports = {registrarUsuario, buscarUsuario, alterarUsuario}



